bind/package.go

```diff
@@ -28,9 +29,17 @@ type Package struct {
 	funcs   []Func
 }
 
+var (
+	mu sync.Mutex
+)
+
 // NewPackage creates a new Package, tying types.Package and ast.Package together.
 func NewPackage(pkg *types.Package, doc *doc.Package) (*Package, error) {
+	// protection for parallel tests
+	mu.Lock()
 	universe.pkg = pkg // FIXME(sbinet)
+	mu.Unlock()
```

Comment:
hum... wouldn't this introduce some inconsistency?

if `test1` enters here first, the `pkg` field points to `pkg1`.
then, the lock is released.
but `test2` could enter, grab the lock, put `pkg2` in the field *before* `test1` finishes to do its thing: `test1` might access `universe.pkg` while `pkg` is in fact `pkg2`.

I'd probably `defer` the call to `mu.Unlock()` to make sure the processing of the `Package` is consistent.

Reply:
That's a fair point- I was really only thinking of protecting the object rather than the impact overall.

Running the tests with a deferred `.Unlock()` encompassing all of `NewPackage()` doesn't seem to take much longer, so that's good. Will update the PR